src/ser/mod.rs

```diff
@@ -453,6 +522,33 @@ mod tests {
     #[test]
     fn str() {
         assert_eq!(&*crate::to_string("hello").unwrap(), r#""hello""#);
+        assert_eq!(&*crate::to_string("").unwrap(), r#""""#);
+
+        // Characters unescaped if possible
+        assert_eq!(&*crate::to_string("ä").unwrap(), r#""ä""#);
+        assert_eq!(&*crate::to_string("৬").unwrap(), r#""৬""#);
+        assert_eq!(&*crate::to_string("\u{A0}").unwrap(), r#"" ""#); // non-breaking space
+        assert_eq!(&*crate::to_string("ℝ").unwrap(), r#""ℝ""#); // 3 byte character
+        assert_eq!(&*crate::to_string("💣").unwrap(), r#""💣""#); // surregate pair
+
+        // " and \ must be escaped
+        assert_eq!(&*crate::to_string("foo\"bar").unwrap(), r#""foo\"bar""#);
```

Comment:
This seems to show no changes, it would be simpler if they were both in raw form...

`to_string(r#"foo"bar"#).unwrap(), r#""foo\"bar""#)`

this makes it clear the escape is added

Reply:
I looked into this again and would like to keep it for consistency. All the tests have a normal Rust string literal on the left, allowing the insertion of unprintable characters or highlighting the unicode code point of the characters. On the right, there is always a result JSON containing quotes.